vcs/local.go

```diff
@@ -0,0 +1,105 @@
+package vcs
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+
+	"golang.org/x/mod/sumdb/dirhash"
+)
+
+func init() {
+	Register(newLocal, "local")
+}
+
+type LocalDriver struct {
+	WatchChanges bool     `json:"watch-changes"`
+	IgnoredFiles []string `json:"ignored-files"`
+}
+
+func newLocal(b []byte) (Driver, error) {
+	d := LocalDriver{}
+
+	if b != nil {
+		if err := json.Unmarshal(b, &d); err != nil {
+			return nil, err
+		}
+	}
+
+	return &d, nil
+}
+
+// Adapted from: https://github.com/golang/mod/blob/ad6fd61f94f8fdf6926f5dee6e45bdd13add2f9f/sumdb/dirhash/hash.go#L44
+func myHash(files []string, open func(string) (io.ReadCloser, error)) (string, error) {
```

Comment:
It looks like this function is identical to the `Hash1` function exposed in dirHash, except yours usds sha1. Is there a reason you didn't go with `Hash1` here? (If there is, I'm probably going to ask that you add it in comment here)

Reply:
I was doing it with some friends on stream. If I recall correctly I didn't use `Hash1` verbatim because it's a base64 string and this doesn't play well with the URLs. The reason for using sha1 was that (supposedly) this is the fastest hash algorithm as implemented in go. Will add some comment!